lib/msal-browser/src/broker/BrokerClientApplication.ts

```
@@ -77,19 +112,64 @@ export class BrokerClientApplication extends ClientApplication {
+                this.logger.info(`Sending auth response: ${brokerAuthResponse}`);
+                clientPort.postMessage(brokerAuthResponse);
+                clientPort.close();
+                return;
+            }
+
             switch (validMessage.interactionType) {
-                case InteractionType.Redirect:
-                    return this.brokeredRedirectRequest(validMessage, clientMessage.ports[0]);
-                case InteractionType.Popup:
-                    return this.brokeredPopupRequest(validMessage, clientMessage.ports[0]);
                 case InteractionType.Silent:
                     return this.brokeredSilentRequest(validMessage, clientMessage.ports[0]);
+                case InteractionType.Redirect:
+                case InteractionType.Popup:
                 default:
-                    return;
+                    const interactionType = this.config.system.brokerOptions.preferredInteractionType && this.config.system.brokerOptions.preferredInteractionType !== InteractionType.None 
+                        ? this.config.system.brokerOptions.preferredInteractionType : validMessage.interactionType;
+                    return this.interactiveBrokerRequest(interactionType, validMessage, clientMessage);
             }
         }
     }
 
+    /**
+     * 
+     * @param interactionType 
+     * @param validMessage 
+     * @param clientMessage 
+     */
+    private async interactiveBrokerRequest(interactionType: InteractionType, validMessage: BrokerAuthRequest, clientMessage: MessageEvent): Promise<void> {
+        switch (interactionType) {
+            case InteractionType.Redirect:
+                return this.brokeredRedirectRequest(validMessage, clientMessage.ports[0]);
+            case InteractionType.Popup:
+                return this.brokeredPopupRequest(validMessage, clientMessage.ports[0]);
+            case InteractionType.Silent:
+            case InteractionType.None:
+                // TODO: Throw error?
```

To answer the question in the todo, yes I'd throw an error here